src/image/loading_displaying.js

```diff
@@ -375,7 +375,7 @@ p5.prototype.saveGif = async function(
         pixels
       );
 
-      data = _flipPixels(pixels);
+      data = _flipPixels(pixels, this.width, this.height);
```

Comment:
Thanks for the fix! Does this still work correctly when you've set `pixelDensity(2)` in `setup`? I wonder if we need to be multiplying by the density here

Reply:
I tried using pixelDensity() in setup but nothing was being rendered. Am I missing something in this code? 

```js
let sketch = function(p) {
  p.setup = function() {
    pixelDensity(2);
    p.createCanvas(500, 500, p.WEBGL);
  };

  p.draw = function() {
  };

  p.keyPressed = function() {
    if (p.key === 's') {
      p.saveGif('test', 5);
    }
  }
};

let myp5 = new p5(sketch);
```

Reply:
In instance mode I think you need `p.pixelDensity(2)`.

Reply:
Oh ya. Thanks!

The current fix works as expected when setting different densities of pixels.

I also checked the behaviour by multiplying the dimensions with density while passing in function call but I was getting another console error saying : ` offset is out of bounds `. So, looks like some calculations go wrong inside flipPixels function in the case when we multiply dimensions with density. 

I saw a similar implementation in readPixelsWebGL() function and stackoverflow answers (https://stackoverflow.com/questions/41969562/how-can-i-flip-the-result-of-webglrenderingcontext-readpixels) so looks like we should not consider multiplying by density here? Or maybe we need to recalculate the flip pixels logic considering density (I tried that but wasn't able to figure out).

Reply:
Ah, wait, logically it should have it, but we just set it to 1 here while exporting: https://github.com/processing/p5.js/blob/58821e2a8b698d1d0b1c90db3b8800a75fcf9809/src/image/loading_displaying.js#L311

So I think width/height on its own is fine!